src/Nethermind/Nethermind.Trie/PatriciaTree.cs

```diff
@@ -646,6 +662,28 @@ void Trace(TrieNode node, in TraverseContext traverseContext)
                 _logger.Trace($"Traversing {node} to {(traverseContext.IsReadValue ? "READ" : traverseContext.IsDelete ? "DELETE" : "UPDATE")}");
             }
 
+            [DoesNotReturn]
+            [StackTraceHidden]
+            static ref readonly CappedArray<byte> TraverseInvalid(TrieNode node)
+            {
+                switch (node.NodeType)
+                {
+                    case NodeType.Branch:
+                        return ref TraverseBranch(node);
+                    case NodeType.Unknown:
+                        return ref TraverseUnknown(node);
+                    default:
+                        return ref ThrowNotSupported(node);
+                }
+            }
+
+            [DoesNotReturn]
+            [StackTraceHidden]
+            static ref readonly CappedArray<byte> TraverseBranch(TrieNode node)
+            {
+                throw new InvalidOperationException($"Branch node {node.Keccak} should already be handled");
```

Comment:
I think its possible that it traverse from branch to extension then back to branch.

Reply:
Have tenary inside `TraverseExtension` that calls `TraverseBranches` direcly so should be ok.

Reply:
For Extension 

Before 

<img width="527" alt="image" src="https://github.com/NethermindEth/nethermind/assets/1142958/23a1d286-84d0-4c73-8d04-9cdb8c01ce51">


After 

<img width="528" alt="image" src="https://github.com/NethermindEth/nethermind/assets/1142958/fc1424f5-7cf1-46c7-bbd4-b934b94dfce2">